test/views/returnFrequency/ChangeReturnFrequencyConfirmationViewSpec.scala

```diff
@@ -26,7 +26,90 @@ class ChangeReturnFrequencyConfirmationViewSpec extends ViewBaseSpec {
 
   "Rendering the Dates Received page for an individual" when {
 
-    "contactPref is 'DIGITAL'" should {
+    "contactPref is 'DIGITAL'" when {
+
+      "emailVerified is true" should {
+
+        lazy val view =
+          views.html.returnFrequency.change_return_frequency_confirmation(contactPref = Some("DIGITAL"), emailVerified = true)(user, messages, mockAppConfig)
+        lazy implicit val document: Document = Jsoup.parse(view.body)
+
+        s"have the correct document title of '${viewMessages.ReceivedPage.title}'" in {
+          document.title shouldBe viewMessages.ReceivedPage.title
+        }
+
+        s"have a correct page heading of '${viewMessages.ReceivedPage.heading}'" in {
+          elementText("#page-heading") shouldBe viewMessages.ReceivedPage.heading
+        }
+
+        s"have the correct h2 '${viewMessages.ReceivedPage.h2}'" in {
+          elementText("h2") shouldBe viewMessages.ReceivedPage.h2
+        }
+
+        s"have the correct p1 of '${viewMessages.ReceivedPage.p1}'" in {
+          paragraph(1) shouldBe viewMessages.ReceivedPage.digiPrefWithEmail
+        }
+
+        s"have the correct p2 of '${viewMessages.ReceivedPage.p2}'" in {
+          paragraph(2) shouldBe viewMessages.ReceivedPage.contactDetails
+        }
+
+        "not have a link to change client" in {
+          elementExtinct("#change-client-text")
+        }
+
+        "have the correct finish button" which {
+
+          s"has the text '${viewMessages.finish}'" in {
+            elementText("#finish") shouldBe viewMessages.finish
+          }
+
+          "has link back to customer details page" in {
+            element("#finish").attr("href") shouldBe mockAppConfig.manageVatUrl
+          }
+        }
+      }
+
+      "emailVerified is false" should {
+
+        lazy val view = views.html.returnFrequency.change_return_frequency_confirmation(contactPref = Some("DIGITAL"))(user, messages, mockAppConfig)
```

Comment:
As the only thing changing between this and the above scenario is the email preference, I would bother retesting all the other features of the page - only the content regarding their preferences